code/modules/power/cell.dm

```diff
@@ -11,15 +11,24 @@
 	throw_speed = 2
 	throw_range = 5
 	w_class = WEIGHT_CLASS_SMALL
-	var/charge = 0	// note %age conveted to actual charge in New
+	/// note %age conveted to actual charge in New
+	var/charge = 0	
 	var/maxcharge = 1000
 	materials = list(/datum/material/iron=700, /datum/material/glass=50)
 	grind_results = list(/datum/reagent/lithium = 15, /datum/reagent/iron = 5, /datum/reagent/silicon = 5)
-	var/rigged = FALSE	// true if rigged to explode
-	var/chargerate = 100 //how much power is given every tick in a recharger
-	var/self_recharge = 0 //does it self recharge, over time, or not?
+	/// true if rigged to explode
+	var/rigged = FALSE	
+	///how much power is given every tick in a recharger
+	var/chargerate = 100 
+	///does it self recharge, over time, or not?
+	var/self_recharge = FALSE
+	///stores the chargerate to restore when hit with EMP, for slime cores
+	var/emp_chargerate_save
+	///Time in SECONDS for EMPs so that recharging cells know when to start charging again. 
+	var/emp_timer = 0
```

Comment:
Rather than using a timer that needs updating every tick, use a world.time counterdown.
There are some defines for cooldown_initialize, but its pretty simple to write one quickly anyway.

Simply have the timer represent a world time of how long the battery should be disabled until. If the current world time is less than that value, do nothing.

To set the battery on a cooldown you can do emp_timer = world.time + 30 SECONDS or whatever you want the cooldown to be.
To check if the battery is still on cooldown you can do
```
if(emp_timer < world.time)
  return
```